flink-java/src/main/java/org/apache/flink/api/java/typeutils/GenericTypeInfo.java

```diff
@@ -66,6 +71,9 @@ public boolean isKeyType() {
 
 	@Override
 	public TypeSerializer<T> createSerializer() {
+		if(useAvroSerializer) {
+			return new AvroSerializer<T>(this.typeClass);
+		}
 		return new KryoSerializer<T>(this.typeClass);
```

Comment:
I respect [yho](http://www.urbandictionary.com/define.php?term=Yho&defid=5440750) and update the PR.